app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Api::V1::LikesController < ApplicationController
+  before_action :set_like, only: %i[ show update destroy ]
+  include Authentication
+
+  # GET /likes
+  def index
+    likes = Like.all
+
+    render json: likes
+  end
+
+  # GET /likes/1
+  def show
+    render json: like
+  end
+
+  # POST /likes
+  def create
+    if authenticate
+      post = Post.find(params[:post_id])
+      user = User.find_by(username: request.headers[:username])
+      like = Like.new(post: post, user: user)
```

Comment:
can like be queried just using the post id, instead of fetching the entire post object first?
try avoiding unnecessary DB calls

Reply:
Got it

Reply:
This piece of code requires me to pass object to create relation. Adding just post_id shows this exception

`"exception": "#<ActiveRecord::AssociationTypeMismatch: Post(#1229460) expected, got \"6\" which is an instance of String(#2220)>"`